core/src/com/unciv/logic/map/TileInfo.kt

```diff
@@ -319,22 +317,29 @@ open class TileInfo {
             tileUniques += city.getMatchingUniques(UniqueType.StatsFromObject, stateForConditionals)
             for (unique in localUniqueCache.get("StatsFromTilesAndObjects", tileUniques)) {
                 val tileType = unique.params[1]
-                if (tileType == improvement) continue // This is added to the calculation in getImprovementStats. we don't want to add it twice
-                if (matchesTerrainFilter(tileType, observingCiv)) 
-                    stats.add(unique.stats)
-                if (tileType == "Natural Wonder" && naturalWonder != null && city.civInfo.hasUnique(UniqueType.DoubleStatsFromNaturalWonders)) {
+                // First check is just a small optimization, as we don't test improvements here.
+                // I'm uncertain as to whether this is worth the time, as string comparison is slow,
```

Comment:
Choose any of the most popular values of tyleType here and it will be an optimization since `matchesTerrainFilter ` does N comparison of strings and having checked the only one the most popular will give its benefits.